config/config_test.go

```diff
@@ -105,5 +121,81 @@ func TestReadConfig(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestParseChainParams(t *testing.T) {
+	var parseChainParamTests = map[string]struct {
+		chainName   []string
+		chainParams chaincfg.Params
+	}{
+		"main network": {
+			[]string{"mainnet"},
+			chaincfg.MainNetParams,
+		},
+		"regtest": {
+			[]string{"regtest"},
+			chaincfg.RegressionNetParams,
+		},
+		"simnet": {
+			[]string{"simnet"},
+			chaincfg.SimNetParams,
+		},
+		"testnet3": {
+			[]string{"testnet3"},
+			chaincfg.TestNet3Params,
+		},
+		"undefined": {
+			[]string{},
+			chaincfg.MainNetParams,
+		},
+		"empty": {
+			[]string{""},
+			chaincfg.MainNetParams,
+		},
+	}
+	for testName, testData := range parseChainParamTests {
+		t.Run(testName, func(t *testing.T) {
+			// use a string builder and a single-value list to represent optionality
```

Comment:
I'm curious as to whether or not you like this potentially-empty-list approach as a way to handle optionality.

The other main option that I can see is splitting the empty option off into its own test case. Thoughts?

Reply:
I like this solution, although I have to admit that I was confused at the beginning and had to figure out what's happening. It may be enough to enhance the comment clarifying more details of this approach. 👌🏻